WEB: Adding new pandas website

datapythonista opened the pull request:
The website can be seen here: https://datapythonista.github.io/pandas-web/

It's still pending that the designer polish the styles. And also the content can surely be improved with everybody's feedback. But I think will be useful to merge this once we iterate a bit on people's feedback, so we can start working on integrating the docs.

I'm coordinating with the devs of numpy/scipy, scikit-learn and matplotlib to see if we can reshare things implemented here. Ideally I'd like to have the `pysuerga.py` script in a separate repo (that's why it's not named pandas_web.py`), and also the layout and pages that can be shared (the code of conduct, the donate, the blog,...). For now I add everything here for simplicity, and will create a new project and have it as a dependency here if they are interested.

CC: @pandas-dev/pandas-core

Commit:
Adding new pandas website

Commit:
Fixing CI

topper-123 commented:
Looks good. Very clean layout.

A few comments:
1: The links to the docs don't work. That's perhaps in the works?

2: Is TideShift a Sponsor of Pandas now? Then they should be on the "with the support of..." list?

3: The link to Wes McKinney's book is http://www.kqzyfj.com/click-7040302-11260198?url=http%3A%2F%2Fshop.oreilly.com%2Fproduct%2F0636920050896.do&cjsku=0636920023784. Is there a reason for going through that ``http://www.kqzyfj.com`` link and not directly to oreilly.com? Looks a bit fishy going through redirects for web shopping, IMO.

Commit:
Adding Tidelift to partners

Commit:
Fixing Wes book link and changing image to the 2nd edition

datapythonista commented:
Thanks for the review @topper-123, good feedback. I added Tidelift to the list of partners, and the docs links not working is intentional. I'm planning to add them after this is merged, and I make the docs use this same layout. Now they'd simply link to the current docs, and you'd leave the website.

Regarding Wes book, I guess that redirect was to keep track on the number of clicks. But besides looking fishy, clicking on it asks me to confirm that I accept some weird conditions, and after that it takes me to the home page of O'Reilly, not the book page (I guess that's because the 2nd edition has been released)_.

I updated the link to the direct link to the second edition, and also change the image to be the second edition front page. @wesm if you want to track clicks to the book, I think the best is to create a page in our own website that performs an automatic redirect, and track the visits with Google Analytics. Let me know if you want me to implement that.

wesm commented on web/pandas/index.html:
```diff
@@ -0,0 +1,157 @@
+                </div>
+				<h4>Get the book</h4>
+				<p>
+					<a href="http://shop.oreilly.com/product/0636920050896.do">
```
If you want to change this to an affiliate link (so we receive ad fees), here is the one: https://amzn.to/2KI5JJw. I can donate annual proceeds from this link (probably not a lot in aggregate) to NumFOCUS

datapythonista replied:
That makes sense, thank you. Updated.

Commit:
Updated link to Wes book with Amazon affiliate link

mroeschke commented:
From a mobile perspective,  most everything rendered okay.

One consistent thing I noticed was that text was a _little_ small for all pages:

![Screenshot_20190819-181603](https://user-images.githubusercontent.com/10647082/63369224-57302700-c334-11e9-9159-317b691cd406.png)

Also this page didn't render too well:

![Screenshot_20190819-181430](https://user-images.githubusercontent.com/10647082/63369351-99596880-c334-11e9-862c-da47a6836057.png)


Wouldn't say these are huge priorities, but keeping it on the radar.

datapythonista commented:
Thanks for the feedback @mroeschke. I'll let the designer from Indeed take care of that. The website uses bootstrap, and should be very easy to make it responsive and look perfect in mobile. But I don't know exactly how it works, so I'll let her do it.

Commit:
Removing carousel and self-hosted juniper.js, replaced Ursa Labs logo by its svg format

Commit:
Merge remote-tracking branch 'upstream/master' into pandas_web

datapythonista commented:
Removed the carousel, and made couple of changes to reduce the size of the website files. There are couple of things I want to change to reduce the size further. After that, if there is no particular change requested, I think we should merge and continue with follow ups. This way we can work in parallel with the integration with the docs and the CI to display the dev version of the web.

The pending changes are:
- Change Anaconda logo to a svg version (there are eps files that can be converted in https://www.anaconda.com/media-kit/)
- Convert to jpeg and increase the compression of the files in the installation page. If anyone wants to propose changes to that page, now it's a good time, before we increase the size of the repo with unnecessary files.

Commit:
Merge remote-tracking branch 'upstream/master' into pandas_web

Commit:
Updating feeds with pandas only posts (Wes and Marc)

Commit:
Merge remote-tracking branch 'upstream/master' into pandas_web

Commit:
Renaming the pandas builder, since for now doesn't seem like any other project is interested in contributing it upstream

Commit:
Making the website responsive (adding viewport meta)

Commit:
Improving styles, and temporary removing the blog and the try/binder page

TomAugspurger reviewed (commented):
Is the new  `web/pandas/static/img/pandas.svg ` the same as `doc/logo/pandas_logo.svg`?

jorisvandenbossche reviewed (commented):
SOrry for the delay, finally took a closer look!

Some general content-wise comments:

- The governance documents currently live in pandas-dev/governance repo. Personally, instead of including it on the website literally in the community section, I think it would be fine to keep it in the separate repo, and link to that from a page on the website about the governance of the project (similarly as is done now: https://pandas.pydata.org/about.html, although the content there can certainly improve :-)) 
- I would personally prefer to keep the development docs in the actual documentation, and not website. Of course, we can still have a high-level page on the main website, but that could then refer to the docs for all details. 
  One reason is why we are building the new docs theme, to have a better navigation. Yes, in theory we can use those templates on the website as well, but that will mean duplication as the theme for the docs has some sphinx specifics in it that will not perfectly match with the jinja templates for the website. 
  If the reason to have it in the website is to always show the latest version, I think we can try to solve this differently (eg linking to the dev version for the contributing section, even from the stable docs; providing a highlight box to link to the dev version on those pages ..)
- Also for the ecosystem page we should not have it duplicated (but I have less an opinion about this one). 

In any case, we should decide on this before merging I think, and depending on what decision remove the other content to not have those things duplicated. 
(short term, to get the basics merged and further iterate, we could also leave them out here from this PR).

Some more specifics on the website:

- If find the current home page somewhat sparse. For example, we should mention that we are open source. And I think some more explanation of what pandas is could be added. 
- A more extensive "About" page can be on the header for me (so it is not hidden under Community, although there is of course also a link to it from the)
- We should probably mention NumFOCUS more clearly on the home page? (eg dask puts it in the footer: https://dask.org/)
- We can probably put some link to github somewhere on the home page

Things like the above are probably not important to fix for this first PR?

And then a technical question:

- you are currently using a custom written `pandas_web.py`. How do you see the advantage of using "our own static site generator" compared to going with an established one as hugo, pelican, jekyll, ... I suppose the main reason is that those are "bloated", since you were able to write what we need in not that many lines of python code (impressive!). And that this one can be very specialized for our site? 
  On the other hand, it might be that going forward we might start replicating features the others already have? (note I have not much experience with static site generators)

datapythonista committed:
Merge remote-tracking branch 'upstream/master' into pandas_web

datapythonista committed:
Adding NumFOCUS pandas posts to the feed

datapythonista committed:
Fixing style/layout of team members

datapythonista committed:
Replacing Anaconda logo by its svg

datapythonista committed:
Fixing nested list

datapythonista committed:
Applying new changes from copied pages (contributing and ecosystem)

datapythonista committed:
Adding that pandas is open source to the short description

datapythonista committed:
Remove link to interactive try pandas page in install, since the page is temporary removed

datapythonista committed:
blackify code

datapythonista committed:
Making the existing logo a link

datapythonista committed:
Reducing the size of the install page images

datapythonista committed:
Fixing image in installation page

datapythonista committed:
Reducing size of book cover

datapythonista committed:
Removing logo and letting choose from the url hash

datapythonista committed:
Minor improvements to the styles

datapythonista committed:
Better quality image of the book

datapythonista committed:
Fixing bug on ignoring files

datapythonista committed:
Improving logo tester

datapythonista commented on web/pandas/community/contributing/index.md:
```diff
@@ -0,0 +1,24 @@
+- [Setting up a development environment](setup.html)
+- [Working with git](git.md)
+- [Contributing to the documentation](docs.html)
+  - [Docstring guidelines](docstring_guidelines.html)
```
Looks like markdown requires 4 spaces not 2, fixed

datapythonista commented on web/pandas/community/contributing/setup.md:
```diff
@@ -0,0 +1,159 @@
+    to use 4 spaces for tabs.
+
+---
+The steps below will download around 900Mb for the pandas
```
Looks like it's 150 Mb for the repo, and 74 for conda dependencies. Not sure why those were wrong (that part is from the worldwide sprint). Fixed

datapythonista commented on web/pandas/community/ecosystem.md:
```diff
@@ -0,0 +1,366 @@
+- ``vaex.from_pandas``
+- ``vaex.to_pandas_df``
+
+## Data validation
```
Updated, just couple of changes in the copied files.

datapythonista commented on web/pandas/_templates/layout.html:
```diff
@@ -0,0 +1,85 @@
+                <a href="#">Back to top</a>
+            </p>
+            <p>
+                © 2009 - 2019, pandas team
```
I rendered it from Windows, not sure why it didn't render it correctly, but the source is correct.

datapythonista commented on web/pandas/install.md:
```diff
@@ -0,0 +1,31 @@
+environment to use pandas. Other installation options can be found in
+the [advanced installation page](https://pandas.pydata.org/pandas-docs/stable/install.html).
+
+If you want to try pandas before installing it locally, you can do it online [here](try.html).
```
Good catch, removed.

datapythonista commented:
Made all the pending updates. The changes add 324Kb to the repo, I don't think this can be reduced more (I reduced the size of every image where it made sense).

I added a temporary hack to visualize how the logos look like:
- http://datapythonista.github.io/pandas-web/#dataframe
- http://datapythonista.github.io/pandas-web/#panda
- http://datapythonista.github.io/pandas-web/#lines

Also, if you want to try a different logo, you can provide the url to the image after the hash: http://datapythonista.github.io/pandas-web/#https://kunlabori.es/wp-content/uploads/2014/12/python-madrid.png

I addressed most of the reported issues, and also improved few styles, thanks all for the feedback.

Comments on the remaining topics:

- **Governance**: I don't see what's the advantage of a separate repo. The governance content here is partially generated from the info in `config.yml`, which is also reused in other parts. I do see a clear advantage of having the info just in one place, so it's not outdated as it happens often with the governance repo. What is the advantage of a separate repo?

- **Development/contributing docs**: The reason why I moved it to the web is because we're not interested in the version of the last release (e.g. `0.25.1`) but instead the latest (`master`). Having it in the website makes this happen automatically. Having it in the docs requires some dirty hacks I think. That's also the reason why I finally discontinued the idea of building the whole website with Sphinx. How would you address this if you have it in the docs, do you have a solution?

- **Custom static site generator**: Your guess is right, I think the advantage of not using Pelican or Hugo is simplicity and flexibility. The static site generator part is around 120 lines of code (the rest are things like fetching the maintainers data from github, which we need anyway). I like Pelican, but besides the added complexity it has limitations that sometimes require complicating things significantly when trying to do simple things. I think `pandas_web.py` is very simple, and adapted exactly to what we need (I don't think we need to add anything else).

- **Home page, footer, about us, github link...**: All those sound good, but I think they make more sense in follow ups. If there can be a quick agreement in any of them I'm happy to implement it here. But surely the home page, and probably the others, they'll be better addressed with an independent PR, so we can have a focused discussion.

WillAyd commented:
On the topic of a separate repo and a custom site generator do we know what this would add in terms of complexity to our documented build process? I think the current site is relatively simple and generated via make rules with how's its set up in a separate repo, so would ideally like to have a drop in replacement for that generation if not already included

datapythonista commented:
Not sure I understand you @WillAyd. The generation of the website takes around 7 seconds (most of the time is spent fetching the maintainers data from github). It just adds a `./pandas_web.py pandas` to the process.

I think in terms of complexity this is simpler than using Pelican, and adding a make file here would make things more complex. Also, I think having a directory web instead of a separate repo make things simpler (and I'd say it'd help avoid that we forget to update things in the separate website, which happened some times with the current website).

Can you please clarify what you mean, not sure what you're proposing.

datapythonista commented:
@jorisvandenbossche probably worth having a look at this when you have time, and see if we can merge the website soon (even a minimal version), so we can work on the integration of the web and docs, and then continue from there. I think the earlier we have the merge the better, so we don't risk having to change too much stuff later on.

TomAugspurger commented:
What are our thoughts on when pandas.pydata.org will be switched over? Coincident with the 1.0 release?

datapythonista commented:
> What are our thoughts on when pandas.pydata.org will be switched over? Coincident with the 1.0 release?

Coincident with 1.0 sounds good to me. Also ok to publish the new website earlier when it's ready, if people prefer that.

TomAugspurger commented:
We'll certainly want to continuously deploy somewhere, for ease of
inspecting changes.

But announcing it to the world with 1.0 sounds good.

On Mon, Sep 16, 2019 at 12:01 PM Marc Garcia <notifications@github.com>
wrote:

> What are our thoughts on when pandas.pydata.org will be switched over?
> Coincident with the 1.0 release?
>
> Coincident with 1.0 sounds good to me. Also ok to publish the new website
> earlier when it's ready, if people prefer that.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/28014?email_source=notifications&email_token=AAKAOIT7QDTC5FE4L5JZHB3QJ63XTA5CNFSM4INCLXP2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6Z2FHA#issuecomment-531866268>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKAOIR7XI2BY6QZSRO5FPLQJ63XTANCNFSM4INCLXPQ>
> .
>

datapythonista commented:
@pandas-dev/pandas-core let me know if there is any other comment here, otherwise I'll merge in the next hours, so I can move forward with the rest of the tasks for the new website (add build to the CI, integrate with docs, open discussion about the home page...)

jorisvandenbossche commented:
For the governance document and contributing docs, I would suggest to leave them out for now, and add them back in a PR, so we can have a focused discussion on those in a separate PR. Otherwise I want to have that discussion here.

jorisvandenbossche commented on web/pandas/_templates/layout.html:
```diff
@@ -0,0 +1,95 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>pandas</title>
```
```suggestion
        <title>pandas - Python Data Analysis Library</title>
```

jorisvandenbossche commented on web/pandas/_templates/layout.html:
```diff
@@ -0,0 +1,95 @@
+              $('[data-toggle="popover"]').popover()
+            })
+            $(function () {
+                // temporary functionality to test different logos
```
Remove this before merging?

jorisvandenbossche commented on web/pandas/_templates/layout.html:
```diff
@@ -0,0 +1,95 @@
+                crossorigin="anonymous"></script>
+        <script>
+            $(function () {
+              $('[data-toggle="tooltip"]').tooltip()
```
Can you add a comment for what those functions are needed?

jorisvandenbossche commented on web/pandas/_templates/layout.html:
```diff
@@ -0,0 +1,95 @@
+                $(".navbar-brand img").attr("src", img);
+                $(".navbar-brand img").height(30);
+            });
+        </script>
```
To add to a todo list: add google analytics

Commit:
Removing governance from new website

jorisvandenbossche commented on web/pandas/community/citing.md:
```diff
@@ -0,0 +1,46 @@
+            year={2011}
+        }
+
+## Brand and logo
```
Once we have a new logo, this probably deserves it's own apge

Commit:
Removing contributing documentation

Commit:
Merging from master

jorisvandenbossche commented on web/pandas/community/license.md:
```diff
@@ -0,0 +1,31 @@
+# License
+
```
For me, this doesn't add much value as separate page on the website, I would say

jorisvandenbossche reviewed (commented):
In addition to the inline comments, the pandas_fibonnaci image can be removed I think?

jorisvandenbossche commented:
I am trying it out locally, but the images are not working for me. Changing `/static/...` to `static/..` in the paths (eg on the main page) seems to fix this, but not sure it is because I am doing something wrong.

Also the links in the navbar don't work locally. 
Is this maybe because I am just opening the html files in the browser (as I am used to with sphinx), and I rather need to serve them on a localhost?

--

Can you add a README in the `web/` directory shortly stating the purpose of that directory?

Commit:
Addressing review comments

datapythonista commented:
Addressed comments.

You need to run `cd build && python -m http.server` to navigate the website, links and image paths are absolute, not relative. If you want to build the website to live in a directory, you need the `--base-url=/directory-for-the-web`.

jorisvandenbossche commented:
> You need to run cd build && python -m http.server to navigate the website ...

Thanks! Can you add that note to the readme / command help?

jorisvandenbossche commented on web/pandas/_templates/layout.html:
```diff
@@ -0,0 +1,85 @@
+                    <span class="navbar-toggler-icon"></span>
+                </button>
+
+                <a class="navbar-brand" href="{{ base_url }}/"><img alt="" src="{{ base_url }}{{ static.logo }}"/></a>
```
this is not working at the moment (I think static.logo is not defined right now?)

Commit:
Adding note about serving the website using an http server

jorisvandenbossche commented:
Another thing: the salsalabs donate container is not working (not locally, but also not on your hosted version)

Commit:
Hidding logo if not present, and removing last screenshot of the installation instructions

datapythonista commented:
The iframe works for me, not sure if it was a temporary problem in their side, or you've got an adblocker or something preventing it to render.

datapythonista commented:
I didn't want to include the logo, to avoid increasing the size of the repo with something will be replaced soon. I'm removing the logo from the html for now, so it doesn't show broken.

jorisvandenbossche commented:
> I didn't want to include the logo, to avoid increasing the size of the repo with something will be replaced soon. I'm removing the logo from the html for now, so it doesn't show broken.

Linking to the logo file in the docs does not work? (and I suppose on build copying that to the build directory)

datapythonista commented:
That logo has the font in black, I could link it, but I'd have to change the header to white to look ok.

I think we still need several iterations with the website until we want to publish it to production, so I would focus on having a starting point so we can open PRs to it, and not in trying to have something final (it's far from it anyway, starting by the home page).

jorisvandenbossche commented on web/pandas/_templates/layout.html:
```diff
@@ -0,0 +1,85 @@
+            (function() {
+                var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
+                ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
+                var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
```
Can you use the same one as for the docs? 
https://github.com/pandas-dev/pandas/blob/372a9a03608f61e4472f2ba0f4b52956f8e1cce1/doc/source/themes/nature_with_gtoc/layout.html#L98-L106

(or is there a reason the above is better? then we can update the docs ones)

Also, can this go at the bottom scripts? Or is there a good reason to load this first?

datapythonista replied:
I copied the one from the website, didn't know the docs had a different one.

I followed Google instructions on where to place it, they say immediately after the `<head>` tag.

jorisvandenbossche replied:
I recently update the docs one, it was on my to do list to also update the website but didn't get to that yet ... (kind of proving your point here :-))

jorisvandenbossche commented:
Yeah, no problem for the logo.

Sorry for more comments, but some final things on duplicated content: the new website also duplicates now the roadmap and the ecosystem pages from the docs. For those I would do the same as for the governance / contributing docs: leave that for separate PRs, as I don't think we should duplicate content here. 
As you said, better to merge this quickly and iterate, so focusing on the infrastructure and not too much on those content details that might need some more discussion makes that easier.

datapythonista commented:
I think there is agreement to move the roadmap and the ecosystem here, so I'll get rid of the versions in the docs if this is now ready to be merged.

jorisvandenbossche commented:
> I think there is agreement to move the roadmap and the ecosystem here,

Where has that been discussed? (I maybe missed some issues .. that's a problem with our huge number of issues)

datapythonista commented:
There are few issues with discussions about the website, but the most important one was in this PR.

If I also remove these pages, this PR contains literally a home page that is incomplete, and not much more. So, I start to miss the point on whether this is a step forward, or we're just wasting time implementing something there is no minimum agreement on.

I'll leave this on hold until those decisions are made, working on this for weeks, and reverting everything again doesn't seem to be a great investment of my time, and it's quite frustrating.

jorisvandenbossche commented:
Sorry, Marc, I didn't want to demean your efforts, and I understand your frustration as I have been very slow with feedback. Sorry for your frustration.

For me, there is still a clear added value of this PR: it is providing the base for working on a better home page (which is great work, as this is highly needed). The exact items in the community dropdown are IMO not essential in the idea of a better website (although I have a strong opinion on it). It was in that light that I thought it would be easier to leave that for later. But I am also happy with merging this as is. It's just some text files that we can also further discuss after they are merged.

WillAyd commented:
I’m also okay with merging as is and whittling down details later. I think this is a great improvement over current configuration 

Sent from my iPhone

> On Sep 17, 2019, at 8:56 AM, Joris Van den Bossche <notifications@github.com> wrote:
> 
> Sorry, Marc, I didn't want to demean your efforts, and I understand your frustration as I have been very slow with feedback. Sorry for your frustration.
> 
> For me, there is still a clear added value of this PR: it is providing the base for working on a better home page (which is great work, as this is highly needed). The exact items in the community dropdown are IMO not essential in the idea of a better website (although I have a strong opinion on it). It was in that light that I thought it would be easier to leave that for later. But I am also happy with merging this as is. It's just some text files that we can also further discuss after they are merged.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

TomAugspurger commented:
Either order works for me as well. We'll be fixing things up anyway (since
we're not yet removing the old roadmap.rst, right?)

So merge this and let's iterate on it over the next few weeks?

On Tue, Sep 17, 2019 at 11:16 AM William Ayd <notifications@github.com>
wrote:

> I’m also okay with merging as is and whittling down details later. I think
> this is a great improvement over current configuration
>
> Sent from my iPhone
>
> > On Sep 17, 2019, at 8:56 AM, Joris Van den Bossche <
> notifications@github.com> wrote:
> >
> > Sorry, Marc, I didn't want to demean your efforts, and I understand your
> frustration as I have been very slow with feedback. Sorry for your
> frustration.
> >
> > For me, there is still a clear added value of this PR: it is providing
> the base for working on a better home page (which is great work, as this is
> highly needed). The exact items in the community dropdown are IMO not
> essential in the idea of a better website (although I have a strong opinion
> on it). It was in that light that I thought it would be easier to leave
> that for later. But I am also happy with merging this as is. It's just some
> text files that we can also further discuss after they are merged.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub, or mute the thread.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/28014?email_source=notifications&email_token=AAKAOISVRYRSTKSB3GWQNK3QKD7HHA5CNFSM4INCLXP2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD65CJUA#issuecomment-532292816>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKAOISSYZM34PUCPB6EUDLQKD7HHANCNFSM4INCLXPQ>
> .
>

jorisvandenbossche commented:
Yes, fine with merging as is and iterate.

---

To try to explain a bit more the potential reason for some of the disagreements we are having (what I also tried to say at EuroScipy regarding the doc theme / integration with the website): 

I have been working on a new documentation theme, and for me, the main driver for that is because I find the *navigation* functionality of the current theme lacking. 
But the current proposal of the website here (and the same for the current homepage) has even less navigation structure compared to the current doc theme. But to be clear, I think this is good for the main website. Because fancy overview landing pages like the home page or an about or donation page don't need much navigational structure. I see the main website mainly as this: attractive overview pages that give a good intro and overview and then further link to more detailed content (docs, github, mailing list, ...). But that also means that I personally prefer that all pages which are more heavily text-content based (like the contributing pages) stay in the documentation (sphinx) section, as there we are have the layout for handling such content. 
Of course, nothing prevents us (technically) to use the same navigation structure (eg the sidebar(s)) also in the website pages. But I *think* this will give some duplicated content for the html layouts (the doc theme still is tailored towards sphinx), and to me it also seems a potential benefit to keep all the pages that uses this navigation in the same place (the sphinx content) while having the other content in the markdown (web) content. But for sure, this is only one option. We can also discuss and experiment with how to re-use some of the navigation structures from the doc theme also on the website.

datapythonista commented:
Thanks for clarifying your point, I didn't understand the reasoning.

The division that I used was what needs to be served from master, and what from the latest release. I think it simplifies significantly how things are built and deployed.

In any case, I think we all agree we need to build the new website in iterations. I think after merging this (I'll do it tomorrow if there are no objections), and making a first integration with the docs, will be easier to see the advantages and disadvantages of every option.

@jorisvandenbossche do you think you can have a first version (possibly minimal) of the new docs layout merged soon? So we can experiment using the same templates in both the website and the docs (I think it shouldn't be difficult, and very useful).

TomAugspurger commented:
Nice work Marc!

On Wed, Sep 18, 2019 at 8:25 AM Marc Garcia <notifications@github.com>
wrote:

> Merged #28014 <https://github.com/pandas-dev/pandas/pull/28014> into
> master.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/28014?email_source=notifications&email_token=AAKAOIVSDRIEOQFLK3RADBDQKIT35A5CNFSM4INCLXP2YY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOTWIKTQI#event-2643503553>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKAOIVMKHZE7IVPHBH2HO3QKIT35ANCNFSM4INCLXPQ>
> .
>
